### ISSUE TITLE
Touch Mode Optimizations

### ISSUE THREAD

**Issue by mweech** [fgpv-vpgf/fgpv-vpgf#1241]
### Goal

Perform optimization related to interaction with the viewer through a touch screen interface.

### Touch Mode
An optional setting in the left menu panel that can be toggled on/off by any user.

**Defaults** to `on` if the `navigator.userAgent` string indicates either `Android|webOS|iPhone|iPad|iPod|BlackBerry|IEMobile|Opera Mini`, `off` otherwise. While this is not a comprehesive check for all types of phones/tablets, it does target a large portion of the userbase.

### Implementation

- [x] increase spacing around buttons and their icons (including legend entry flags) when:
    - [x] ~~the layout is small or;~~
    - [x] touch mode is on

- [x] increase size of the smallish buttons (buttons in map nav, zoom-to and details buttons in the datatable)
- [x] increase size of the legend items

- [x] ~~fix legend scrolling as dragula immediately starts reorder by starting reorder only on long press. Suggestion is to cancel mouse down timeout on scroll events and mouse movement events~~ (new issue created #1343)

- [x] Set navbar controls (zoom in/out, basemap select etc) to fully opaque for improved visibility

- [x] show basemap description on click, not on hover (no hover event on mobile devices) PR #1263

- [x] Only display a tooltip during a long press. When released the tooltip is hidden. A long press occurs after 0,75 second.

- [x] ~~zoom to feature on small/medium layouts doesn't seem to do anything when you press it. Make the datatable panel (and any underlaying) panels almost transparent like we do for basemaps so that the user can see something has happened. Restore the opacity on the next tap~~ (new issue created #1344)

**Comment by milesap** [comment]
Two good articles on this: 

- [Touch and Mouse](https://www.html5rocks.com/en/mobile/touchandmouse/)
- [You Can't Detect A Touchscreen](http://www.stucox.com/blog/you-cant-detect-a-touchscreen/)

tldr; touchscreen detection is flawed and should be avoided. Often touchscreen optimizations are good for mouse yielding citizens as well (i.e. bigger buttons).

**Comment by milesap** [comment]
Will assign estimate and start work once design discussion is complete (most likely 1PM tomorrow as scheduled)

**Comment by milesap** [comment]
Starting work on this issue.

Brief pod discussions with @AleksueiR, we will display a `touch mode` toggle option in the left menu at all times and for all screen sizes. We will perform some limited detection for the most widely used devices like iPhone, iPad, Android . . . and if detected automatically toggle `touch mode` on. 

Estimate increased to 8 after reviewing this issue in more detail.

@mweech target release version may need to be bumped to `v1.5.0`, I'll leave it at `v1.4.0` for now.

**Comment by milesap** [comment]
See closed PR #1347
